lazyflow/utility/roiRequestBuffer.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, Dict, Iterator, Tuple, Type
+from queue import Queue
+
+import numpy
+
+from lazyflow.graph import Slot
+from lazyflow.utility.roiRequestBatch import RoiRequestBatch
+
+
+ROI = Tuple[int, int, int, int, int]
+SHAPE = ROI
+ROI_TUPLE = Tuple[ROI, ROI]
+
+
+class _RoiIter:
+    """Iterate over rois for a given shape"""
+
+    def __init__(self, shape: SHAPE):
+        self._shape = shape
+
+    def to_index(self, roi: ROI_TUPLE) -> int:
+        """Convert ROI_TUPLE to index
+
+        first ROI_TUPLE -> index = 0
+        last ROI_TUPLE -> index = len(self) - 1
+        """
+        return numpy.ravel_multi_index(roi[0][:-2], self._shape[:-2])
+
+    def __len__(self) -> int:
```

Comment:
I guess this assumes shapes are in `xyztc` order like the rest of the codebase... Do you think we should move into something with less magical constants in this PR? I'm super biased here, but I'd strongly argue for using `Shape5D` and `Slice5D`everywhere instead of just `Tuple[int, int, int, int, int]`. And there's even the `Slot.getShape5D()` method to help with that. In fact, we could use `Slice5D.split()` to iterate over tiles of any shape

Reply:
Since `ndstructs`' API is still not ready for prime time, maybe you could use a dict here, which is consistent with `MetaDict.getTaggedShape()`? It'd be easier to understand that `roi[0][:-2]` means "just the spacial axes" if it were using the label names instead of their indices

Reply:
it's actually `ctz`.... proves your point I guess :)

Reply:
I'm expecting this code path to be gone soon. The RoiIter thing will now raise if axisorder is not as expected `tzcyx`.